packages/filebrowser-extension/src/index.ts

```diff
@@ -691,6 +693,85 @@ export const launcherToolbarButton: JupyterFrontEndPlugin<void> = {
   }
 };
 
+/**
+ * A plugin to open files from remote URLs
+ */
+const openUrlPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/filebrowser-extension:open-url',
+  autoStart: true,
+  requires: [IFileBrowserFactory, ITranslator],
+  optional: [ICommandPalette],
+  activate: (
+    app: JupyterFrontEnd,
+    factory: IFileBrowserFactory,
+    translator: ITranslator,
+    palette: ICommandPalette | null
+  ) => {
+    const { commands } = app;
+    const trans = translator.load('jupyterlab');
+    const { defaultBrowser: browser } = factory;
+    const command = CommandIDs.openUrl;
+
+    commands.addCommand(command, {
+      label: args =>
+        args.path ? trans.__('Open %1', args.path) : trans.__('Open from URL…'),
+      caption: args =>
+        args.path ? trans.__('Open %1', args.path) : trans.__('Open from URL'),
+      execute: async args => {
+        let url: string | undefined = (args?.url as string) ?? '';
+        if (!url) {
+          url =
+            (
+              await InputDialog.getText({
+                label: trans.__('Url'),
+                placeholder: 'https://example.com/path/to/file',
+                title: trans.__('Open Url'),
```

Comment:
We should probably use `URL` instead of `Url`

```suggestion
                label: trans.__('URL'),
                placeholder: 'https://example.com/path/to/file',
                title: trans.__('Open URL'),
```